src/test/isolation2/input/resgroup/resgroup_move_query.source

```diff
@@ -141,9 +145,150 @@ SELECT is_session_in_group(pid, 'rg_move_query_mem_small') FROM pg_stat_activity
+1<:
+1: SELECT gp_inject_fault('resource_group_give_away1', 'resume', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2<:
+2: SELECT num_running FROM gp_toolkit.gp_resgroup_status WHERE rsgname='rg_move_query';
+
+-- test9: check destination group has no slot leaking if move signal processed at the time target process became dead
+-- sleep at first process
+-- start to move it at second process, but suspend before sending signal to it
+-- wait until first process finish and exit
+-- resume at second process, it should throw an error
+1: SET ROLE role_move_query_mem_small;
+1&: SELECT pg_sleep(3);
+2: SELECT gp_inject_fault('resource_group_give_away1', 'reset', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2: SELECT gp_inject_fault('resource_group_give_away1', 'suspend', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2&: SELECT gp_toolkit.pg_resgroup_move_query(pid, 'rg_move_query') FROM pg_stat_activity WHERE query LIKE '%pg_sleep%' AND rsgname='rg_move_query_mem_small';
+1<:
+1q:
+3: SELECT gp_inject_fault('resource_group_give_away1', 'resume', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2<:
+2: SELECT num_running FROM gp_toolkit.gp_resgroup_status WHERE rsgname='rg_move_query';
+
+-- test10: check destination group has no slot leaking if we got an error on latch waiting
+-- sleep at first process
+-- start to move it at second process, send moving signal to first process
+-- suspend at first process after handling moveto* params
+-- throw an error in WaitLatch block at second process, this will force pg_resgroup_move_query to continue
+-- second process should throw an error, but consider moveto* params handled by target
+-- resume at first process
+-- first process should continue with moving as all slot control is on it's side
+1: SET ROLE role_move_query_mem_small;
+1: SELECT gp_inject_fault('resource_group_move_handler2', 'reset', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+1: SELECT gp_inject_fault('resource_group_move_handler2', 'suspend', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+1: BEGIN;
+1&: SELECT pg_sleep(5) FROM gp_dist_random('gp_id');
+2: SELECT gp_inject_fault('resource_group_give_away2-1', 'reset', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2: SELECT gp_inject_fault('resource_group_give_away2-1', 'suspend', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2: SELECT gp_inject_fault('resource_group_give_away2-2', 'reset', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2: SELECT gp_inject_fault('resource_group_give_away2-2', 'error', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2&: SELECT gp_toolkit.pg_resgroup_move_query(pid, 'rg_move_query') FROM pg_stat_activity WHERE query LIKE '%pg_sleep%' AND rsgname='rg_move_query_mem_small';
+3: SELECT pg_sleep(3);
```

Comment:
Instead of `pg_sleep()` call we might use multiple `gp_wait_until_triggered_fault()` ones to wait for passing of all interesting fault points.

Reply:
Fixed in all tests where it was possible.